### PR TITLE
fix R16 support on velvet and new_hash macro

### PR DESCRIPTION
**Pull request by kuenishi**
From [Kelly's patch](https://gist.github.com/kellymclaughlin/b27755fc6eda6b55c373) , with  Stanchion ( https://github.com/basho/stanchion/pull/64 ) and velvet ( https://github.com/basho/velvet/pull/12 ).

CC: https://github.com/basho/velvet/pull/11

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+    case erlang:system_info(otp_release) =< "R15B01" of
+        true ->
+            CONFIG;
+        false ->
+            HashDefine = [{d,new_hash}],
+            case lists:keysearch(erl_opts, 1, CONFIG) of
+                {value, {erl_opts, Opts}} ->
+                    lists:keyreplace(erl_opts,1,CONFIG,{erl_opts,Opts++HashDefine});
+                false ->
+                    [{erl_opts, HashDefine}]
```
Comment (kellymclaughlin) [inline rebar.config.script]: This line needs to be:

```
CONFIG ++ [{erl_opts, HashDefine}]
```

**Comment by kellymclaughlin** [comment]
`make compile-riak-test` yields this error output:

```
==> riak_cs (riak_test_compile)
riak_test/src/rtcs.erl:150: syntax error before: '.'
riak_test/src/rtcs.erl:263: syntax error before: '.'
riak_test/src/rtcs.erl:296: syntax error before: '.'
riak_test/src/rtcs.erl:420: syntax error before: '.'
riak_test/src/rtcs.erl:438: syntax error before: '.'
riak_test/src/rtcs.erl:476: syntax error before: '.'
riak_test/src/rtcs.erl:481: syntax error before: '.'
riak_test/src/rtcs.erl:486: syntax error before: '.'
riak_test/src/rtcs.erl:491: syntax error before: '.'
riak_test/src/rtcs.erl:543: syntax error before: '.'
riak_test/src/rtcs.erl:550: syntax error before: '.'
riak_test/src/rtcs.erl:60: function deploy_nodes/2 undefined
riak_test/src/rtcs.erl:84: function deploy_nodes/2 undefined
riak_test/src/rtcs.erl:145: function riak_oss_config/1 undefined
riak_test/src/rtcs.erl:188: function riak_oss_config/1 undefined
riak_test/src/rtcs.erl:341: function set_admin_creds_in_configs/3 undefined
riak_test/src/rtcs.erl:362: function start_stanchion/0 undefined
riak_test/src/rtcs.erl:363: function start_cs/1 undefined
riak_test/src/rtcs.erl:366: function start_cs/1 undefined
riak_test/src/rtcs.erl:372: function stop_cs/1 undefined
riak_test/src/rtcs.erl:373: function stop_stanchion/0 undefined
riak_test/src/rtcs.erl:378: function stop_cs/1 undefined
riak_test/src/rtcs.erl:387: function start_stanchion/0 undefined
riak_test/src/rtcs.erl:388: function start_cs/1 undefined
riak_test/src/rtcs.erl:393: function start_cs/1 undefined
riak_test/src/rtcs.erl:399: function stop_cs/1 undefined
riak_test/src/rtcs.erl:400: function stop_stanchion/0 undefined
riak_test/src/rtcs.erl:407: function stop_cs/1 undefined
riak_test/src/rtcs.erl:618: crypto:sha_mac/2 is deprecated and will be removed in in a future release; use crypto:hmac/3
ERROR: riak_test_compile failed while processing /Users/kelly/basho/repos/riak_cs: rebar_abort
make: *** [compile-riak-test] Error 1
```

**Comment by kellymclaughlin** [comment]
I have a fix for the above that I will just push to this branch. I'm +1 for merge with the addition of that change.

**Comment by kuenishi** [comment]
Thank you for adding r_t stuff!

**Pull request merged**